planning/ThingDescription/td-next-work-items/usability-and-design.md

```diff
@@ -63,12 +63,19 @@ Related Issues:
 
 Also known as: Mapping TD elements to messages
 
-How can we represent all protocol messages sent and received?
-It should be using data schema to indicate application-specific parameters but we should be able to describe payload, header, uri variables, and other protocol parameters.
+- Some protocols have multiple side-channels, e.g HTTP has Headers, URI-Variables, Payload/Body.
+- Some protocols may have a default/unique `contentType` for at least one of the channels, others are flexible. e.g. HTTP Headers are key-value strings.
+- Some protocols do not have the concept of `contentType` at all.
 
-Related Issues:
+#### Open questions
+How can we describe how the logical information (state for property, input/output messages for actions, messages for event) is mapped to each channel?
+Do we need to reconsider how we express `contentType` and `contentCoding` in the case more than a side-channel is flexible?
+Do we want to provide a way to express a serialization scheme for protocols that are binary-only?
 
+#### Related Issues:
 - BACnet URI Variables discussion
+- Modbus description of complex data types
+- jsonrpc-over-websocket
```

Comment:
```suggestion
- [jsonrpc-over-websocket](https://github.com/w3c/wot-binding-templates/issues/125)
```

I am not sure if you have this one in mind, that is why a separate suggestion.

Reply:
The issue is good, in general might be a good idea to cover jsonrpc-over-websocket and socketio-over-websocket, assuming we have people already using it.